components/ai_chat/core/common/mojom/ai_chat.mojom

```diff
@@ -567,6 +570,12 @@ interface ConversationUI {
   OnConversationDeleted();
 };
 
+interface Metrics {
+  WillSendPromptWithFullPage();
+
+  WillSendPromptWithQuickAction();
+};
```

Comment:
The functions sound like an observer but the name sounds like a service - which is it? Is this something notifying the metrics service that something is about to happen, in which case perhaps better names will be `RecordSendPromptXYZ`, or is it something subscribing to notifications, in which case `MetricsObserver` might be a clearer name?

Reply:
Aside from naming, suggest adding a comment for this interface, would help make it clear too.

Reply:
> The functions sound like an observer but the name sounds like a service - which is it?

It's an interface for `AIChatMetrics`, to be called from webui

> Is this something notifying the metrics service that something is about to happen, in which case perhaps better names will be RecordSendPromptXYZ

A `Record` prefix may be misleading, since the methods don't actually record any metrics. Other methods in this class prefixed with `Record` actually do record a metric, so I'd prefer to keep this `WillSend`-prefixed naming scheme (or a very similar one) for consistency.

Reply:
Renamed with `On` prefix